### DISCUSSION TITLE
how to do, or new feature?

### DISCUSSION THREAD

**Discussion by XSTechnology** [arendst/Tasmota#16028] [Q&A] [answered]
what I'm trying to do with version 12.

on / off pulsed signal (~500Milsec)  to control latching relays which switch 120v lights.
Relays have 4 wires (24v+ latch on) (24v+ latch off)

I have 25 relays in house wired with all ceiling lights, wall switches (push button) are only switching 24volts to turn on/off the lights.
everything 24v comes down to a central panel in the basement.
Wired it this way in 1984 when I built the house.
At that time plan was to control with XT computer, relay board and voice recognition software.........
Life happens and fast forward to today.

using esp32, and heavily modified Fauxmo (multi tasking, asyc tcp / udp)
I have some coding running, interfaced with Alexa.
That works, but I would still need to add syslog, Mq reporting, etc.

Current config. is on/off control of lights mcp23017 16 relay board, switching 24v with banks 1-8  switch on banks 9 - 16 switch off.

After an hour of config work, I have 16 relays visible and controlled by Alexa. The web portal shows 16 on/off switched.
I can set the power option to pulse 500 milsec. , however the power status is reported back to alexa as off, after the pulse.

What I need help with is:

I need the  8 devices supported as an on and off bank of 8 relays each.
The power status needs to be reflected as on or off based on commands received, not status of the relay.
So.. I think I need a virtual device, separate from the actual state of the relay.

**Comment by barbudor** [comment, marked as answer]
Hi
I don't see another way than virtual relays for that
Unfortunately Tasmota doesn't natively have that but you can either:
- use gpios defined as Relay but actually not connected
- use more MCP outputs for the same
- as you are using ESP32 you can use Berry to create yourself a driver  for virtual relays that will not need any actual pin. You may get inspiration from the Shift595 sample code here https://github.com/arendst/Tasmota/blob/development/tasmota/berry/drivers/Shift595.be
